core/services/bag_of_holding/main.py

```diff
@@ -0,0 +1,89 @@
+#! /usr/bin/env python3
+import json
+import logging
+from pathlib import Path
+from typing import Any, Dict
+
+import appdirs
+import uvicorn
+from commonwealth.utils.apis import GenericErrorHandlingRoute
+from commonwealth.utils.logs import InterceptHandler, init_logger
+from fastapi import FastAPI
+from fastapi.responses import HTMLResponse, JSONResponse
+from fastapi_versioning import VersionedFastAPI, version
+from loguru import logger
+from pydantic import BaseModel
+
+SERVICE_NAME = "bag_of_holding"
+FILE_PATH = Path(appdirs.user_config_dir("bag of holding", "db.json"))
+
+logging.basicConfig(handlers=[InterceptHandler()], level=0)
+init_logger(SERVICE_NAME)
+
+app = FastAPI(
+    title="Bag of Holding API",
+    description=(
+        "Bag of Holding implements a FastAPI service with versioning that provides a simple key-value"
+        "storage API, enabling the user to store and retrieve data as JSON objects through HTTP requests."
+    ),
+)
+app.router.route_class = GenericErrorHandlingRoute
+logger.info("Starting Bag of Holding")
+
+app = FastAPI()
+
+
+class KeyValue(BaseModel):
+    key: str
+    value: Any
+
+
+def read_db() -> Any:
+    try:
+        with open(FILE_PATH, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except FileNotFoundError:
+        return {}
+
+
+def write_db(data: Dict[str, Any]) -> None:
+    with open(FILE_PATH, "w", encoding="utf-8") as f:
+        json.dump(data, f)
+
+
+@app.get("/get/{key}")
+@version(1, 0)
+async def get_value(key: str) -> JSONResponse:
+    db = read_db()
+    if key in db:
+        return JSONResponse(content={"value": db[key]})
+    return JSONResponse(content={"error": "Key not found"}, status_code=404)
+
+
+@app.post("/set")
+@version(1, 0)
+async def set_value(item: KeyValue) -> JSONResponse:
+    db = read_db()
+    db[item.key] = item.value
+    write_db(db)
+    return JSONResponse(content={"status": "success"})
+
```

Comment:
I would recommend to allow reading and setting values from nested paths:

```python
# Check https://pypi.org/project/dpath/

@app.post("/write/{path}")
@version(1, 0)
async def write_data(path: str, data: DataModel):
    current_data = read_db()
    dpath.util.new(current_data, path, data.data)
    write_db(current_data)
    return JSONResponse(content={"status": "success"})

@version(1, 0)
@app.get("/get/{path}")
async def read_data(path: str):
    current_data = read_db()
    try:
        result = dpath.util.get(current_data, path)
    except KeyError:
        raise HTTPException(status_code=400, detail="Invalid path")
    return JSONResponse(content={"value": db[key]})
```

Reply:
great idea!